pkg/ccl/sqlproxyccl/connector.go

```diff
@@ -118,9 +122,16 @@ func (c *connector) OpenTenantConnWithToken(
 	// only return once we've seen a ReadyForQuery message.
 	//
 	// NOTE: This will need to be updated when we implement query cancellation.
-	if err := readTokenAuthResult(serverConn); err != nil {
+	newBackendKeyData, err := readTokenAuthResult(serverConn)
+	if err != nil {
 		return nil, err
 	}
+	func() {
+		c.CancelInfo.mu.Lock()
+		defer c.CancelInfo.mu.Unlock()
+		c.CancelInfo.mu.origBackendKeyData = newBackendKeyData
```

Comment:
There is a small race condition here, but it is probably  fine. The race is this:
thread 1 > The sql proxy begins processing a transfer request.
thread 2 > The sql proxy receives a cancel request and starts sending it to the backend.
thread 1 > The sql proxy finishes the transfer request and starts forwarding to the new server.
thread 2 > The cancel is processed by the original sql server. The cancel is dropped since the connection does not exist anymore.

Reply:
i don't believe this has any impact, since a transfer request can't be initiated if there are any running queries. (and also i believe no new queries can start once the transfer request begins)